opt_adapt/opt.py

```diff
@@ -171,16 +269,18 @@ def minimise(
         dJ_ = None if it == 1 else op.dJdm_progress[-1]
         if order == 1:
             args = (u_plus, u_, dJ_)
+        elif order == 2:
+            args = (u_plus, u_, dJ_, B)
         else:
-            raise NotImplementedError(f"Only order 1 methods are supported, not {order}")
+            raise NotImplementedError(f"Method {method} unavailable")
 
         # Take a step
         cpu_timestamp = perf_counter()
         out = {}
         for o in step(it, forward_run, mesh, params, *args, Rspace=Rspace):
             out.update(o)
         J, u, dJ = out["J"], out["u"], out["dJ"]
-        lr, u_plus, u_, dJ_ = out["lr"], out["u+"], out["u-"], out["dJ-"]
+        lr, u_plus, u_, dJ_, B = out["lr"], out["u+"], out["u-"], out["dJ-"], out["B"]
```

Comment:
I would not do it this way. Instead, we should *not* return `B` from gradient descent (because it doesn't make sense) and here there should be
```
if order > 1:
    B = out["B"]
```
on the next line.